### ISSUE TITLE
7.1 (4) breaks encryption in MUC

### ISSUE THREAD

**Issue by jtmaston** [tigase/siskin-im#164]
**Describe the bug**
Not sure why, but updating to 7.1(4) on Testflight breaks encryption in MUC. Avatars are not displayed, only the first letter and messages all read "message decryption failed". Outgoing messages are unaffected, they get sent correctly.

**To Reproduce**
Update
Go to MUC
???
Not profit

**Expected behavior**
Messages should be sent, avatars shown.
**Screenshots**
Don't think these are needed...

**Details (please complete the following information):**
 - Siskin Version: [e.g. 7.1.0(4)]
 - Branch: [e.g. TestFlight]
 - iOS version [e.g. 15.2.1]
 - iPhone model [e.g. iPhone 11]

**Additional context**

**Comment by licaon-kter** [comment]
Is 4 the latest build? Try to update, retest...

**Comment by jtmaston** [comment]
I should have mentioned, the bug persists even in 7.1(7), which is the latest.

**Comment by hantu85** [comment]
The version which was released around 20 minutes ago?

Also, please keep in mind that the issue you are reporting is with other clients sending messages to Siskin which is unable to decode. 

What clients are being used for sending those messages? 
Are you using other clients on the same account?
If you are using other clients, is your SiskinIM-OMEMO key enabled?

**Comment by jtmaston** [comment]
Yes, both the latest version ( the one from half an hour ago ) and the one before it have this issue.

We have a MUC hosted on a Prosody server.

I can send messages to private convos without issues.

My friends are both using Blabber.im clients.

I use BeagleIM as my other client, and it works without issue. Of note is that messages I send from Beagle are readable in siskin and vice versa.

I have my keys trusted.

**Comment by hantu85** [comment]
SiskinIM and BeagleIM both are using the same library for OMEMO encryption which was tested against Monal and Conversations and was working fine. Actually, after changes in version 7.1 (b1-b7), it is even more stable than it was before.

I cannot comment on Blabber.IM as I'm not using it (no Android device laying around), but you've mentioned that you are not seeing their avatars. That would mean, that either their presence is not forwarded properly or there is some issue with iq forwarding between Blabber.IM clients and your clients (iq forwarding done by MUC in Prosody).

**Comment by jtmaston** [comment]
Hm is there any chance I might find something in the app logs?

**Comment by centertivevier** [comment]
Hi, 

a friend from me with the following device:

-     Siskin Version: 7.1
-     Branch: Stable (from 14.02.2022)
-     iOS version 14.4
-     iPhone 7

has the same problem. Sending in Mucs with Omemo is all fine, but when they receiving messages with Omemo, he can´t read it. We have tested with Dino 0.3 and Conversations 2.9.10. The problem is since update to the new version. Anyone a idea?

They would be update in first step the IOS Version and give you an update, if they help.

**Comment by jtmaston** [comment]
Yeah, I've tried changing server's ( using the tigase xmpp server also doesn't seem to solve the issue ), it's still there...

**Comment by hantu85** [comment]
@jtmaston What you have actually changed? used new account? or created a new room on Tigase XMPP Server MUC?

**Comment by centertivevier** [comment]
@hantu85. a little more  information from my side. We used a Ejabberd 21.12 and i have create a new muc with my friend, but there is the same problem. He can send with omemo, but received can't read "message decryption failed".

**Comment by jtmaston** [comment]
Thanks, I'll have a look when that drops.
Sadly it's been pretty hectic on my end so I haven't had more time to test.

When I deployed tigase I made a whole new user, new MUC, had a friend create a user and join the MUC from his client and the issue was still there. 

I'll come back with more testing soon

**Comment by hantu85** [comment]
Could someone check if the issue happens (recipient not able to decrypt) when the sender is in the room or only when the sender of the encrypted message is not in the room when the recipient actually received this message?

**Comment by sokai** [comment]
> Could someone check if the issue happens (recipient not able to decrypt) when the sender is in the room or only when the sender of the encrypted message is not in the room when the recipient actually received this message?

I had the situation (while writing the last comment) that all Siskin users (and me, writing test messages ;-)) were in the room.

**Comment by licaon-kter** [comment]
As said, my users see the same. Conversations users are always online, as usual.

**Comment by hantu85** [comment]
Did anyone check when this error message occurs, if on the list of occupants in the room in SiskinIM senders for which decryption of messages fails are visible as available?

**Comment by hantu85** [comment]
Newly released beta build 7.2-b1 (on TestFlight), contains a fix on recently reported issue with participant status presentation in MUC rooms. This may be cause of this issue as well.

If possible, please try new build and check if it solved the issue. If not, please report error code presented in the error message, which will help us fix the issue.

**Comment by sokai** [comment]
@hantu85 I got the positve feedback from an iOS user that the recent TestFlight version is working (again)! #thanks!

So can you please publish the TestFlight version as the current/stable release.(?)

**Comment by sokai** [comment]
@jtmaston Is the problem fixed, now with 7.2? 🤔
Thanks + KR!

**Comment by jtmaston** [comment]
> @jtmaston Is the problem fixed, now with 7.2? 🤔
> 
> Thanks + KR!

Sadly, I have no means to test it now.
Xmpp proved to be too unreliable for our application so we switched to something else.
